tensorflow_datasets/image/cartoonset.py

```diff
@@ -0,0 +1,150 @@
+          description="A collection of random, 10000 2D cartoon avatar images",
+          version="0.1.0",
+          data='cartoonset10k',
+      ),
+      CartoonsetConfig(
+          name='cartoonset100k',
+          description="A collection of random, 100000 2D cartoon avatar images",
+          version="0.1.0",
+          data='cartoonset100k',
+      )
+  ]
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict(
+            {"image": tfds.features.Image(shape=_CARTOON_IMAGE_SHAPE),
+             "eye_angle": tfds.features.ClassLabel(num_classes=3),
+             "eye_lashes": tfds.features.ClassLabel(num_classes=2),
+             "eye_lid": tfds.features.ClassLabel(num_classes=2),
+             "chin_length": tfds.features.ClassLabel(num_classes=3),
+             "eyebrow_weight": tfds.features.ClassLabel(num_classes=2),
+             "eyebrow_shape": tfds.features.ClassLabel(num_classes=14),
+             "eyebrow_thickness": tfds.features.ClassLabel(num_classes=4),
+             "face_shape": tfds.features.ClassLabel(num_classes=7),
+             "facial_hair": tfds.features.ClassLabel(num_classes=15),
+             "hair": tfds.features.ClassLabel(num_classes=111),
+             "eye_color": tfds.features.ClassLabel(num_classes=5),
+             "face_color": tfds.features.ClassLabel(num_classes=11),
+             "hair_color": tfds.features.ClassLabel(num_classes=10),
+             "glasses": tfds.features.ClassLabel(num_classes=12),
+             "glasses_color": tfds.features.ClassLabel(num_classes=7),
+             "eye_slant": tfds.features.ClassLabel(num_classes=3),
+             "eyebrow_width": tfds.features.ClassLabel(num_classes=3),
+             "eye_eyebrow_distance": tfds.features.ClassLabel(num_classes=3)}),
+        supervised_keys=None,
+        # Homepage of the dataset for documentation
+        urls=["https://google.github.io/cartoonset/"],
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    # There is no predefined train/val/test split for this dataset.
+    path = dl_manager.manual_dir
+    if not tf.io.gfile.exists(path):
+      msg = 'You must download the dataset files manually and place them in: '
+      msg += ', '.join([path])
+      raise AssertionError(msg)
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=2,
+            gen_kwargs={
+                "filepath": path
+            })
+    ]
+
+  def _generate_examples(self, filepath):
+    """Yields examples."""
+    for path, _, files in tf.io.gfile.walk(
+        os.path.join(filepath, self.builder_config.name)):
+      for file in files:
+        features_dict = dict()
+        name, dtype = file.split('.')
+        if dtype == 'png':
+          image = tfds.core.lazy_imports.skimage.io.imread(
```

Comment:
you have to read using tf.gfile (to be compatible with non-local filesystems):

          with tf.io.gfile.GFile(os.path.join(root, fname), "rb") as png_f:
            mask = tfds.core.lazy_imports.cv2.imdecode(
                np.fromstring(png_f.read(), dtype=np.uint8), flags=0)